agent/auto-config/auto_config.go

```diff
@@ -392,6 +392,20 @@ func (ac *AutoConfig) resolveHost(hostPort string) []net.TCPAddr {
 // This will persist the configuration to disk (unless in dev mode running without
 // a data dir) and will reload the configuration.
 func (ac *AutoConfig) recordAutoConfigReply(reply *agentpb.AutoConfigResponse) error {
+	// overwrite the auto encrypt DNS SANs with the ones specified in the auto_config stanza
+	if len(ac.config.AutoConfig.DNSSANs) > 0 && reply.Config.AutoEncrypt != nil {
+		reply.Config.AutoEncrypt.DNSSAN = ac.config.AutoConfig.DNSSANs
+	}
+
+	// overwrite the auto encrypt IP SANs with the ones specified in the auto_config stanza
+	if len(ac.config.AutoConfig.IPSANs) > 0 && reply.Config.AutoEncrypt != nil {
```

Comment:
I don't know about AuthEncyrpt, why does this need to be overwritten?

Reply:
This configuration could be provided on the client. If there is client specific configuration like `DNSSAN`, it shouldn't be overwritten by the server configuration.

Reply:
Right now instead of generating TLS certificates and pushing them down (which we will transition to doing at some point hopefully soon), the servers push down configuration that will enable auto_encrypt for generating the TLS certificates.

The AutoConfig settings allow specifying IP/DNS SANs and the AutoEncrypt settings do too. When auto_encrypt is being enabled by auto_config we copy over the requested IP/DNS SANs from the AutoConfig settings to the AutoEncrypt settings.